### ISSUE TITLE
Cannot terminate application

### ISSUE THREAD

**Issue by Mantriur** [k9mail/k-9#1078]
There is no way to simply end K9. This function is needed to:

1. Prevent unwanted network use
2. Prevent the battery drain when not needed

Great, app, but having to use the clunky android settings to terminate the app is a big downer.

**Comment by cketti** [comment]
See https://github.com/k9mail/k-9/wiki/Manual-globalsettings#background-sync

In the future please use the [mailing list](http://groups.google.com/group/k-9-mail) for support requests.

**Comment by Mantriur** [comment]
Sorry, this should have been marked as a feature request. It's not about changing the applications behavior, but terminating the application. Unless that setting is sufficient to prevent the app from ever waking up without user interaction. It's not just about network traffic, it is also (and in my case foremost) about battery life, and a little bit about the requirement of having full control over the software running on a sensitive device with a mediocre security track record.

**Comment by cketti** [comment]
If there's nothing to do, the app won't run in the background.

Adding an option to quit the app won't happen. See also http://blog.radioactiveyak.com/2010/05/when-to-include-exit-button-in-android.html

**Comment by Mantriur** [comment]
I don't even know where to begin.

1. There is an undeniable difference between the back and an exit button. The back button is ambiguous, but the general expectation is that whatever is reasonable to do in the background may be done, while an exit button is a clear signal by the user to stop all activity, e.g. K9 fetching emails. Unlike the back button, it does not leave any room for interpretation.

2. Assuming that one knows better than the user is the original sin of UI design. The users knows best, however idiotic his approach may be. It's the difference between a config file and a UI.

3. The examples named are all Google apps. Yes, Google would like apps to never really quit, because they have a vested financial interest in their own apps always being permitted to collect the data that's at the core of their business model. Google is heavily biased and I assure you, your goals do not align.

4. It is completely irrelevant what "most" apps technically do with their exit button. An exit button is a promise: You promise that your application will stop doing anything. If it keeps doing stuff, it's malware. If you're not doing anything anyway, great, then use the same stuff for back and exit. In the case of K9 it would need to stop fetching emails until started again. Offering an option, which disables the only thing done in the background, is not very useful for people not intimately familiar with the application. This level of knowledge can not be expected, especially not for every individual application.

5. The technical process is also irrelevant. It is nice that Android gives such fine grained control over the current state of the application, I'm all for using that. It does not, however, override the need for an exit button, because none of these signals is a definite kill request, neither are they interpreted as such in practice.

6. None of his arguments describe a satisfactory solution for a user's desire to have the security of an application not executing any further. He patronizes the users with the claim that an exit button would not be properly implemented by most devs anyway. He does however provide a lengthy description of the way it's supposed to work, which is far likelier to cause total ambiguity, and does not offer any transparency or guarantee to the user whatsoever. From a journalistic point of view, this is an opinion piece, and, I'm sorry to say, the opinion is not convincingly expressed.

I know of no valid UXD argument supporting this, other than Google's desire to prevent the user from temporarily withdrawing his consent to data collection, because that would cut into their revenue.

**Comment by Mantriur** [comment]
That being said, it's your app and your choice of course. Just tired of everybody preaching Google's story without reflecting. Sorry for ranting.

**Comment by locutis-of-borg-1999** [comment]
Under general circumstances, K-9 should be allowed "background sync", and then it DOES run in the background, killing the battery if there's no connectivity; EVEN in Airplane Mode.  And, for THAT, there's a system intent that could be read!  
https://developer.android.com/reference/android/content/Intent.html#ACTION_AIRPLANE_MODE_CHANGED  

And as far as security, the "bundled" app's are never a good arguement, despite the radioactiveyak.  Even NPR has bemoaned app's on iPhone official repo's, versus jailbroken wares.  And Google?!?!  Please!!!!!   If "Remote Kill Switch" doesn't speak volumes, nothing will!!!!!  

Bottom line is that K-9 is eating battery, trying ro connect, when there's no path to the internet.
